packaging/linux/fedora/sunshine.spec

```diff
@@ -0,0 +1,252 @@
+  "-DCMAKE_INSTALL_PREFIX=%{_prefix}"
+  "-DSUNSHINE_ASSETS_DIR=%{_datadir}/sunshine"
+  "-DSUNSHINE_EXECUTABLE_PATH=%{_bindir}/sunshine"
+  "-DSUNSHINE_ENABLE_WAYLAND=ON"
+  "-DSUNSHINE_ENABLE_X11=ON"
+  "-DSUNSHINE_ENABLE_DRM=ON"
+  "-DSUNSHINE_PUBLISHER_NAME=LizardByte"
+  "-DSUNSHINE_PUBLISHER_WEBSITE=https://app.lizardbyte.dev"
+  "-DSUNSHINE_PUBLISHER_ISSUE_URL=https://app.lizardbyte.dev/support"
+)
+
+function install_cuda() {
+  # check if we need to install cuda
+  if [ -f "%{_builddir}/cuda/bin/nvcc" ]; then
+    echo "cuda already installed"
+    return
+  fi
+
+  local cuda_prefix="https://developer.download.nvidia.com/compute/cuda/"
+  local cuda_suffix=""
+  if [ "$architecture" == "aarch64" ]; then
+    local cuda_suffix="_sbsa"
+  fi
+
+  if [ "$architecture" == "aarch64" ]; then
+    # TODO: this isn't working... only relevant on fedora 39 for now, since cuda doesn't work on f40+ due to gcc version
+
+    # we need to patch the math-vector.h file for aarch64 fedora
+    # back up /usr/include/bits/math-vector.h
+    math_vector_file="/usr/include/bits/math-vector.h"
+
+    # patch headers https://bugs.launchpad.net/ubuntu/+source/mumax3/+bug/2032624
+    cp "$math_vector_file" "$math_vector_file.bak"
+    sed -i 's/__Float32x4_t/int/g' "$math_vector_file"
+    sed -i 's/__Float64x2_t/int/g' "$math_vector_file"
+    sed -i 's/__SVFloat32_t/float/g' "$math_vector_file"
+    sed -i 's/__SVFloat64_t/float/g' "$math_vector_file"
+    sed -i 's/__SVBool_t/int/g' "$math_vector_file"
+  fi
+
+  local url="${cuda_prefix}${cuda_version}/local_installers/cuda_${cuda_version}_${cuda_build}_linux${cuda_suffix}.run"
+  echo "cuda url: ${url}"
+  wget "$url" --progress=bar:force:noscroll -q --show-progress -O "%{_builddir}/cuda.run"
+  chmod a+x "%{_builddir}/cuda.run"
+  "%{_builddir}/cuda.run" --silent --toolkit --toolkitpath="%{_builddir}/cuda" --no-opengl-libs --no-man-page --no-drm
+  rm "%{_builddir}/cuda.run"
+}
+
+# we need to clear these flags to avoid linkage errors with cuda-gcc-c++
+export CFLAGS=""
+export CXXFLAGS=""
+export FFLAGS=""
+export FCFLAGS=""
+export LDFLAGS=""
+
+if [ -n "$cuda_version" ] && [[ " ${cuda_supported_architectures[@]} " =~ " ${architecture} " ]]; then
+  install_cuda
+  cmake_args+=("-DSUNSHINE_ENABLE_CUDA=ON")
+  cmake_args+=("-DCMAKE_CUDA_COMPILER:PATH=%{_builddir}/cuda/bin/nvcc")
+fi
+
+# cmake
+cd %{_builddir}/Sunshine
+echo "cmake args:"
+echo "${cmake_args[@]}"
+cmake "${cmake_args[@]}"
+make -j$(nproc) -C "%{_builddir}/Sunshine/build"
+
+%check
+# setup Xvfb
+export DISPLAY=:99
+Xvfb ${DISPLAY} -screen 0 1024x768x24 &
+
+# run tests
+cd %{_builddir}/Sunshine/build
+./tests/test_sunshine
+
+# teardown Xvfb
+Xvfb -kill ${DISPLAY}
+
+%install
+cd %{_builddir}/Sunshine/build
+%make_install
+
+# Add modules-load configuration
+install -D -m 0644 /dev/stdin %{buildroot}/usr/lib/modules-load.d/uhid.conf <<EOF
+uhid
+EOF
+
+%post
+# TODO: can we just use postinst script directly?
```

Comment:
unfortunately no, in an `rpm-ostree` environment (which is immutable Fedora) the way that packages get installed into is does not very well with this. I ran into the issue previously and found this way to be the best.